src/dodal/log.py

```diff
@@ -124,28 +126,31 @@ def set_up_file_handler(
 
 def set_up_logging_handlers(
     logging_level: Optional[str] = "INFO",
-    dev_mode: bool = False,
+    dev_mode: Optional[bool] = False,
```

Comment:
Should: I think this typing is wrong. Following it back through the code it looks like it's only ever specified in prod through the command line with a `store_true`, according to https://docs.python.org/3/library/argparse.html#action this is either True or False, I don't see how it could be `None`. Feel free to spin into another issue, in which case remove this optional and the check below.